index.html

```diff
@@ -1246,334 +1098,787 @@ <h3>DID URL Content Metadata</h3>
+				another using the <code>ExampleVerifiedFileService</code> from
+				the previous example for integrity-verified files. The path
+				prefix determines which service to use.
+			</p>
+
+			<pre>
+		// DID URLs:
+		did:web:example.com/path/to/file.txt
+		did:web:example.com/example/resume.pdf/&lt;hash&gt;
+
+		// DID Document service section:
+		{
+		"service": [
+			{
+			"type": "FileService",
+			"serviceEndpoint": "https://example.com/",
+			"prefix": "/"
+			},
+			{
+			"type": "ExampleVerifiedFileService",
+			"serviceEndpoint": "https://example.com/example/files/",
+			"prefix": "/example/"
+			}
+		]
+		}
+
+		// Resulting URLs:
+		https://example.com/path/to/file.txt
+		https://example.com/example/files/resume.pdf (with hash verification after resolution)
+			</pre>
+			</section>
 		</section>
+	</section>
+
+	<section id="dereferencing-algorithm">
+		<h3>DID URL Dereferencing Algorithm</h3>
+		<p>
+			This section defines the normative algorithm for dereferencing a
+			<a>DID URL</a> into a resource. It builds on the rules for <a
+			href="did-url-components">DID URL Components Handling</a> section of
+			this specification.
+		</p>
+		<ol>
+			<li>
+				<strong>Parse the <a>DID URL</a> into the following components:</strong>
+				<ul>
+					<li><code>did</code></li>
+					<li><code>path</code></li>
+					<li><code>query parameters</code> (stored as a working set <code>qp</code>)</li>
+					<li><code>fragment</code></li>
+				</ul>
+			</li>
+
+			<li>
+				<strong>Handle Fragments:</strong>
+				<ul>
+					<li>As noted in <a href="#did-url-fragments">DID URL
+					fragments handling</a> section, if the <a>DID URL</a>
+					contains a fragment, a <a>DID Resolver</a> SHOULD NOT process
+					or interpret it. Fragment handling is the client's
+					responsibility given the resolved document and media
+					type.</li>
+				</ul>
+			</li>
 
+			<li>
+				<strong>Resolve the DID:</strong>
+				<ul>
+					<li>Identify and apply <a href="#did-parameters">DID
+					parameters</a> from <code>qp</code>
+					that affect DID Resolution (e.g., <a href="#versionid-query-parameters">versionId</a>,
+					<a href="#versiontime-query-parameters">versionTime</a>),
+					including any <a>DID Method</a>-specific and DID Document
+					Properties Extensions [[?DID-EXTENSIONS-PROPERTIES]] <a
+					href="#did-parameters">DID parameters</a> understood by the
+					<a>DID Resolver</a>.</li>
+					<li>Use these to resolve the <code>did</code> into a
+					<code>didDocument</code> and
+					<code>didResolutionMetadata</code>.</li>
+					<li>If resolution fails, return an error and abort
+					dereferencing.</li>
+					<li>Remove any consumed <a href="#did-parameters">DID
+					parameters</a> from the
+					<code>qp</code> working set.</li>
+				</ul>
+			</li>
+
+			<li>
+				<strong>No Path Component:</strong>
+				<ul>
+					<li>If no <code>path</code> is present, resolution is
+					complete and the path-handling step is skipped.</li>
+				</ul>
```

Comment:
One-bullet lists generally shouldn't be lists.
```suggestion
				If no <code>path</code> is present, resolution is
				complete and the path-handling step is skipped.
```